### ISSUE TITLE
The Importance of Pretrained Model Weight.

### ISSUE THREAD

**Issue by Zhu-haow** [google-research/vision_transformer#56]
I download the official pretrain weight of R50-ViT-B/16. After load the pretrained weight, I got high accuracy on my dataset. If I drop the pretrained weight, we observe a significant drop in accuracy. This is impressive and explainable because Transformer needs pretrain to work well.

However, my question is: 
When I drop part of the pretrained model weight., for example, the projection 'embedding' layer(between the R50 and Transformer Encoder). I got a significant drop in accuracy too. Which is incredible and hard to explain.

Does anyone have the same problem as me? How can I change part of the pretrain weight but still keep comparable performance as baseline model ?

**Comment by koolvn** [comment]
@Zhu-haow 
> When I drop part of the pretrained model weight., for example, the projection 'embedding' layer(between the R50 and Transformer Encoder)

Actually weights depend on arcitecture, so if you remove any intermediate layer then the rest of the network becomes "useless" and needs to be trained again

**Comment by Zhu-haow** [comment]
In fact, I keep all layers. I just remove a single conv pretrained weight and random initialize this conv again. Nothing changes except this. But when I train this model, I got significant lower accuracy (10 percent).

**Comment by mmiaz** [comment]
@Zhu-haow Hi Zhu, Im new to flax and jax, and am having trouble fine tuning the pre-trained model on self-defined dataset, would you mind sharing how to build self-defined dataset as well as how to do the fine-tuning...

I have tried the command in main page README but the job got killed all the time.. I am kinda stuck now... Would be great to learn from you. Many thanks!

**Comment by Zhu-haow** [comment]
> @Zhu-haow Hi Zhu, Im new to flax and jax, and am having trouble fine tuning the pre-trained model on self-defined dataset, would you mind sharing how to build self-defined dataset as well as how to do the fine-tuning...
> 
> I have tried the command in main page README but the job got killed all the time.. I am kinda stuck now... Would be great to learn from you. Many thanks!

Hope can be helpful. you can use below code to get data
`    all_image_paths = list(data_path.glob('*/*'))
    all_image_paths = [str(path) for path in all_image_paths]  # 所有图片路径的列表
    # random.shuffle(all_image_paths)  # 打散
    #
    # image_count = len(all_image_paths)
    # print(image_count)

    label_names = sorted(item.name for item in data_path.glob('*/') if item.is_dir())

    label_to_index = dict((name, index) for index, name in enumerate(label_names))

    all_image_labels = [label_to_index[pathlib.Path(path).parent.name] for path in all_image_paths]
    ds = tf.data.Dataset.from_tensor_slices((all_image_paths, all_image_labels))`

then use 
`data = data.map(_pp, tf.data.experimental.AUTOTUNE)`
to map your data to preprocess function of _pp
